pkg/cableengine/ipsec/ipsec.go

```diff
@@ -338,6 +338,18 @@ func (i *Engine) InstallCable(endpoint types.SubmarinerEndpoint) error {
 		}
 	}
 
+	// MASQUERADE (on the GatewayNode) the incoming traffic from the remote cluster (i.e, remoteEndpointIP)
+	// and destined to the local PODs (i.e., localSubnet) scheduled on the non-gateway node.
+	// This will make the return traffic from the POD to go via the GatewayNode.
+	for _, localSubnet := range i.LocalSubnets {
+		ruleSpec := []string{"-s", remoteEndpointIP, "-d", localSubnet, "-j", "MASQUERADE"}
+		klog.V(8).Infof("Installing iptables rule for MASQ incoming traffic: %s", strings.Join(ruleSpec, " "))
```

Comment:
A minor thing but the strings.Join(...) format arg carries some expense - we can elide its evaluation by first checking if the log level is enabled:

    if klog.V(8) {
        klog.Infof(...)
    }

Reply:
I tried to follow the convention used in the current API. If we decide to do this change, there are multiple such instances in the same API (like line number 316, 323 etc) where this needs to be changed. If you are Ok, we can do it as a separate patch set.

Reply:
That's fine.